planner/core/logical_plan_builder.go

```diff
@@ -157,7 +157,9 @@ func (b *PlanBuilder) buildResultSetNode(node ast.ResultSetNode) (p LogicalPlan,
 			col.OrigTblName = col.TblName
 			if x.AsName.L != "" {
 				col.TblName = x.AsName
-				col.DBName = model.NewCIStr("")
+				if b.ctx.GetSessionVars().CurrentDB == col.DBName.L {
```

Comment:
Could you add a comment here? Why should we check the currentDB name and set DBName to empty string?

Reply:
We can put an example here.

Reply:
I don't know why we set `col.DBName = model.NewCIStr("")` here @zimulala 
I should remove this line ... but it would make explain result (column name) change and affect many rows in the test

Reply:
OK, I'll remove this line... that's a better fix, anyway

Reply:
@tiancaiamao 
I am not sure, I just changed the name of `DBName`. This line of code was originally written like this.
PTAL @XuHuaiyu

Reply:
I guess the reason may be this:
When a table has an alias name, this "virtual table" does not belong to any database, we thought the DBName should be blank.
But I think it seems ok to keep the DBName here.